src/yaml.ts

```diff
@@ -1,7 +1,7 @@
 import * as yaml from 'js-yaml';
 
-export function loadYaml<T>(data: string, opts?: yaml.LoadOptions): T {
-    return yaml.safeLoad(data, opts) as T;
+export function loadYaml<T = any>(data: string, opts?: yaml.LoadOptions): T | undefined {
+    return (yaml.safeLoad(data, opts) as unknown) as T | undefined;
```

Comment:
Can you please add a test case that tests for these condition changes? I think I understand how this behaves differently but it would be good to add a test to make sure we don't break this behaviour in future.

Reply:
I suppose I could, but it's completely trivial tbh, and the compiler will catch any improper use of the function (which would basically involve not handling the undefined case). Anyway, since this type error may only have come up on my side, possibly due to TypeScript version differences, I can just close the PR in favor of #493.